onadata/libs/utils/logger_tools.py

```diff
@@ -1468,3 +1469,54 @@ def _exec_cached_elist_counter_commit_failover() -> None:
             )
             report_exception(subject, msg)
             cache.set(ELIST_FAILOVER_REPORT_SENT, "sent", 86400)
+
+
+def delete_xform_submissions(
+    xform: XForm,
+    instance_ids: list[int] | None = None,
+    soft_delete: bool = True,
+    deleted_by: User | None = None,
```

Comment:
Can we make deleted_by a required field? We should ensure we track who is making the deletion request.